src/LocalDatastoreController.localStorage.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+/* global localStorage */
+
+const LocalDatastoreController = {
+  fromPinWithName(name: string): ?any {
+    const values = localStorage.getItem(name);
```

Comment:
Shouldn't we use AsyncStorage also, because that's how react native provides it. The localDataStoreCOntroller methods should all be async, this way we can swap LocalStorage or AsyncStorage. Otherwise we'll face issues with RN right away.

If we don't implement the AsyncStorage bridge now, we should still make all those methods async as we know for sure we'll need it in the future. 

Unfortunately, we have to use promises so we don't require babel-regenrator to be integrated in the front end SDK.